### PR TITLE
ENH: Add FIAPARCH volatility model

### PR DESCRIPTION
**Pull request by gjunjie**
Implements the Fractionally Integrated Asymmetric Power ARCH (FIAPARCH) volatility model, partially addressing https://github.com/bashtage/arch/issues/531

FIAPARCH extends FIGARCH with an asymmetry parameter (gamma) and a flexible power parameter (delta). Includes Python/Cython recursions, FIAPARCHUpdater for ARCH-in-mean support, arch_model(..., vol="fiaparch") integration, and tests.

### CODE REVIEW

**Comment by bashtage** [comment]
I've only just skimmed and it looks good on first pass.  Any chance you could hit edge cases that are missed in coverage?

**Comment by gjunjie** [comment]
> I've only just skimmed and it looks good on first pass. Any chance you could hit edge cases that are missed in coverage?

Thanks! @bashtage Let me resolve the comments and think more on the edge cases.

File: arch/univariate/volatility.py [hunk 1]
```diff
@@ -3389,6 +3390,501 @@
...
+                raise ValueError("delta must be between 0.05 and 4")
+            self._delta = delta
+        if p < 0 or q < 0 or p > 1 or q > 1:
+            raise ValueError("p and q must be either 0 or 1.")
+        if o < 0 or o > 1:
+            raise ValueError("o must be either 0 or 1.")
+        if self._truncation <= 0:
+            raise ValueError("truncation must be a positive integer")
+        self._num_params = 2 + p + q + o + int(self._est_delta)
+        self._name = self._generate_name()
```
[the remark this answers is not in the document]
Comment (gjunjie) [reply arch/univariate/volatility.py:3483]: override the default

File: arch/univariate/volatility.py [hunk 2]
```diff
@@ -3389,6 +3390,501 @@
...
+            if not 0.05 < delta < 4:
+                raise ValueError("delta must be between 0.05 and 4")
+            self._delta = delta
+        if p < 0 or q < 0 or p > 1 or q > 1:
+            raise ValueError("p and q must be either 0 or 1.")
+        if o < 0 or o > 1:
+            raise ValueError("o must be either 0 or 1.")
+        if self._truncation <= 0:
+            raise ValueError("truncation must be a positive integer")
+        self._num_params = 2 + p + q + o + int(self._est_delta)
```
[the remark this answers is not in the document]
Comment (gjunjie) [reply arch/univariate/volatility.py:3482]: override the default

File: arch/univariate/recursions_python.py [hunk 3]
```diff
@@ -567,6 +567,81 @@
...
+            shock = abs_resids[t - i - 1] - gamma * resids[t - i - 1]
+            sigma_delta[t] += lam[i] * (shock**delta)
+        sigma2[t] = sigma_delta[t] ** (2.0 / delta)
+        sigma2[t] = bounds_check(sigma2[t], var_bounds[t])
+        sigma_delta[t] = sigma2[t] ** (delta / 2.0)
+
+    return sigma2
+
+
+fiaparch_recursion = jit(fiaparch_recursion_python, nopython=True)
```
[the remark this answers is not in the document]
Comment (gjunjie) [reply arch/univariate/recursions_python.py:642]: fiaparch_recursion is consumed by volatility.py through the rec alias
